crates/storage/src/lazy/mapping.rs

```diff
@@ -85,14 +86,27 @@ where
     where
         Q: scale::EncodeLike<K>,
     {
-        pull_packed_root_opt(&self.storage_key(key))
+        pull_packed_root_opt(&self.storage_key(&key))
+    }
+
+    /// Clears the value at `key` from storage.
+    ///
+    /// Returns the value at the key if it previously existed, otherwise returns `None`.
+    pub fn remove<Q>(&self, key: Q) -> Option<V>
+    where
+        Q: scale::EncodeLike<K>,
+    {
+        let storage_key = self.storage_key(&key);
+        let v = self.get(key)?;
```

Comment:
This always loads the value from storage before clearing it. This is not always needed for all types and thus adds extra overhead and therefore no longer follows the zero-overhead principles. We should make sure to only load data from storage when actually needed.

Reply:
How do you recommend we do this? Is there an easy way to check if a value already exists in storage without loading it?

Reply:
You don't need to clear packet. You only need to do a `ink_env::clear_contract_storage(root_key)`. It will clear storage cell if it is not empty

Reply:
> You don't need to clear packet. You only need to do a `ink_env::clear_contract_storage(root_key)`. It will clear storage cell if it is not empty

Unfortunately this is not entirely correct since there are `T` with `T: PackedLayout` where `PackedLayout` actually does something important upon clean-up. An example for this is the `ink_storage::Box` type.

Reply:
So should we be using `ink_env::get_contract_storage()` instead?

Reply:
https://github.com/paritytech/ink/blob/7432565b7e91cbe41f47be57ee0a344429724cc8/crates/storage/src/traits/mod.rs#L224

Reply:
The `lazy` storage facilities implement exactly this for single cells, take a look:
https://github.com/paritytech/ink/blob/7432565b7e91cbe41f47be57ee0a344429724cc8/crates/storage/src/lazy/lazy_imap.rs#L293

Reply:
I think the approach with `SpreadLayout::REQUIRES_DEEP_CLEANUP` is good. Later we can simplify it during https://github.com/paritytech/ink/issues/906